File: angular/frontend/src/app/classes/comment.ts
import { User } from "./user";

export class Comment {

    id!: number;
    comment_text!: string;
    comment_upvote!: number;
    comment_downvote!: number;
    comment_reply_to!: number;
    comment_date!: Date;
    user!: User;
/*
    constructor(
        id: number,
        comment_text: string,
        comment_upvote: number,
        comment_downvote: number,
        comment_reply_to: number,
        comment_date: Date
        ) {
            this.id = id;
            this.comment_text = comment_text;
            this.comment_upvote = comment_upvote;
            this.comment_downvote = comment_downvote;
            this.comment_reply_to = comment_reply_to;
            this.comment_date = comment_date;
        }*/
    
}

File: spring/artplatform/src/main/java/com/artplatform/artplatform/comment/Comment.java
package com.artplatform.artplatform.comment;

import com.artplatform.artplatform.post.Post;
import com.artplatform.artplatform.user.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Table(name="comment")
@Getter
@Setter
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String comment_text;

    private Integer comment_upvote;

    private Integer comment_downvote;

    private Integer comment_reply_to;

    @CreationTimestamp
    private LocalDateTime comment_date;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name="user_id", nullable = false)
    private User user;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name="post_id", nullable = false)
    private Post post;

}

